src/media-server.js

```diff
@@ -27,36 +24,35 @@ export class MediaServer {
   #serverState
 
   /**
-   * @param {object} params
-   * @param {(projectPublicId: string) => Promise<import('./mapeo-project.js').MapeoProject>} params.getProject
+   * @param {Object} params
    * @param {import('fastify').FastifyServerOptions['logger']} [params.logger]
    */
-  constructor({ getProject, logger }) {
+  constructor({ logger } = {}) {
     this.#fastifyStarted = false
     this.#host = '127.0.0.1'
     this.#port = 0
 
-    this.#fastify = fastify({ logger })
-
-    this.#fastify.register(BlobServerPlugin, {
-      prefix: BLOBS_PREFIX,
-      getBlobStore: async (projectPublicId) => {
-        const project = await getProject(projectPublicId)
-        return project[kBlobStore]
-      },
-    })
-
-    this.#fastify.register(IconServerPlugin, {
-      prefix: ICONS_PREFIX,
-      getProject,
-    })
+    this.#fastify = Fastify({ logger })
 
     this.#serverState = new StateMachine({
       start: this.#startServer.bind(this),
       stop: this.#stopServer.bind(this),
     })
   }
 
+  get [kFastify]() {
+    return this.#fastify
+  }
```

Comment:
wasn't sure how useful it is to expose the fastify instance. maybe it is, maybe it isn't 🤷 (currently not being used anywhere)

Reply:
Can be useful in unit tests using `fastify.inject()`. We don't have those yet though, and if we create the fastify instance outside the class (see suggestion) then not needed.